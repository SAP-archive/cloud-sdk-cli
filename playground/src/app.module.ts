import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';

import { CatalogueModule } from './catalogue/catalogue.module';

@Module({
  imports: [CatalogueModule],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
