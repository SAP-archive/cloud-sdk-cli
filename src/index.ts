/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export { run } from '@oclif/command';
