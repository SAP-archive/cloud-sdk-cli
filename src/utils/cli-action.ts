/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */
import cli from 'cli-ux';

export async function action(output: string, showSpinner: boolean, promise: Promise<any>) {
  if (showSpinner) {
    cli.action.start(output);
  } else {
    cli.log(`${output}...`);
  }
  await promise;
  if (showSpinner) {
    cli.action.stop();
  }
}
