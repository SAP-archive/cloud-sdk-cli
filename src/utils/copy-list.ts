/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export interface CopyDescriptor {
  sourcePath: string | URL;
  targetFolder: string;
  fileName: string;
}
