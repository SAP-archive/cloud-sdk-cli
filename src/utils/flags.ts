/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

import Command from '@oclif/command';
import { OutputFlags } from '@oclif/parser';

type FlagsInput<CommandT extends typeof Command> = CommandT['flags'];

export type Flags<CommandT extends typeof Command> = OutputFlags<NonNullable<FlagsInput<CommandT>>>;
export type RequiredFlagsParam<CommandT extends typeof Command, ParamsT extends keyof Flags<CommandT>> = Pick<Flags<CommandT>, ParamsT>;
export type OptionalFlagsParam<CommandT extends typeof Command, ParamsT extends keyof Flags<CommandT>> = Partial<
  RequiredFlagsParam<CommandT, ParamsT>
>;
export type FlagsParam<
  CommandT extends typeof Command,
  RequiredParamsT extends keyof Flags<CommandT>,
  OptionalParamsT extends keyof Flags<CommandT>
> = RequiredFlagsParam<CommandT, RequiredParamsT> & OptionalFlagsParam<CommandT, OptionalParamsT>;
