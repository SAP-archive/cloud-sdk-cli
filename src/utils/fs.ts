/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */

import { promisify } from 'util';
import * as fs from 'fs';
import * as rimraf from 'rimraf';

export const access = promisify(fs.access);
export const mkdir = promisify(fs.mkdir);
export const readdir = promisify(fs.readdir);
export const readFile = promisify(fs.readFile);
export const copyFile = promisify(fs.copyFile);
export const writeFile = promisify(fs.writeFile);
export const rm = promisify(rimraf);
