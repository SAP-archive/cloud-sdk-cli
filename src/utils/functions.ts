/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */

export interface Confirm {
  confirm: (message: string) => Promise<boolean>;
}

function confirm(message: string): Promise<boolean> {
  return Promise.resolve(true);
}

export const testFunctions = {
  confirm
};
