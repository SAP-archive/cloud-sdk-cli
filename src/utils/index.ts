/*!
 * Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
 */

export * from './copy-list';
export * from './generate-vdm-util';
export * from './git-ignore';
export * from './jest-config';
export * from './manifest-yaml';
export * from './message-formatter';
export * from './package-json';
export * from './scaffold';
export * from './templates';
export * from './usage-analytics';
export * from './warnings';
