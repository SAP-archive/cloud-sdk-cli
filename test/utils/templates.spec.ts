/*!
 * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
 */
import * as fs from 'fs';
import * as path from 'path';
import * as rm from 'rimraf';
import { copyFiles, ensureDirectoryExistence, findConflicts, readTemplates } from '../../src/utils';

const pathPrefix = path.resolve(__dirname, __filename.replace(/\./g, '-')).replace('-ts', '');

function getCleanProjectDir(name: string) {
  const projectDir = path.resolve(pathPrefix, name);
  if (fs.existsSync(projectDir)) {
    rm.sync(projectDir);
  }
  fs.mkdirSync(projectDir, { recursive: true });
  return projectDir;
}

describe('Templates Utils', () => {
  afterAll(() => {
    rm.sync(pathPrefix);
  });

  it('ensure directory existence', () => {
    const projectDir = getCleanProjectDir('ensure-dir-existence');
    const dir = path.resolve(projectDir, 'foo', 'bar');
    const file = path.resolve(dir, 'abc', 'package.json');

    ensureDirectoryExistence(dir, true);
    expect(fs.existsSync(dir)).toBe(true);

    ensureDirectoryExistence(file);
    expect(fs.existsSync(path.resolve(dir, 'abc'))).toBe(true);
  });

  it('should return information which files to copy where', () => {
    const initCopyInfo = readTemplates({
      from: [path.resolve(__dirname, '..', '..', 'src', 'templates', 'init')],
      to: 'abcdef'
    });
    expect(initCopyInfo.map(copyInfo => path.basename(copyInfo.fileName)).sort()).toMatchSnapshot();

    const appRouterCopyInfo = readTemplates({
      from: [path.resolve(__dirname, '..', '..', 'src', 'templates', 'add-approuter')],
      to: 'blablabla'
    });
    expect(appRouterCopyInfo.map(copyInfo => path.basename(copyInfo.fileName)).sort()).toMatchSnapshot();
  });

  it('should find conflicts', async () => {
    const projectDir = getCleanProjectDir('find-conflicts');
    fs.writeFileSync(path.resolve(projectDir, '.npmrc'), 'foobar');
    findConflicts(
      readTemplates({
        from: [path.resolve(__dirname, '..', '..', 'src', 'templates', 'init')],
        to: projectDir
      }),
      true
    );
    expect(fs.existsSync(path.resolve(projectDir, '.npmrc'))).toBe(false);
  });

  it('should copy files locally', async () => {
    const projectDir = getCleanProjectDir('copy-files-locally');
    copyFiles(
      readTemplates({
        from: [path.resolve(__dirname, '..', '..', 'src', 'templates', 'init')],
        to: projectDir
      }),
      {}
    );
    expect(fs.readdirSync(projectDir).sort()).toMatchSnapshot();
  });

  // it('should copy files remotely', async () => {
  //   const projectDir = getCleanProjectDir('copy-files-remotely');
  // });
});
