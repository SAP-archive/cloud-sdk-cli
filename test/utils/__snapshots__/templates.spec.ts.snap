// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates Utils should copy files locally 1`] = `
Array [
  ".npmrc",
  "Jenkinsfile",
  "credentials.json",
  "deployment",
  "manifest.yml",
  "pipeline_config.yml",
  "s4hana_pipeline",
  "systems.json",
]
`;

exports[`Templates Utils should return information which files to copy where 1`] = `
Array [
  ".gitkeep",
  ".gitkeep",
  ".gitkeep",
  ".gitkeep",
  ".gitkeep",
  ".npmrc",
  "Jenkinsfile",
  "credentials.json",
  "manifest.yml",
  "pipeline_config.yml",
  "systems.json",
]
`;

exports[`Templates Utils should return information which files to copy where 2`] = `
Array [
  ".npmrc",
  "manifest.yml",
  "package.json",
  "xs-app.json",
  "xs-security.json",
]
`;
